### PR TITLE
feat(1Anime): Add presence

### PR DESCRIPTION
**Pull request by faetalize**
## Description 

Initial commit for a 1anime presence (Support play/pause state, current anime being watched and episode count).
## Acknowledgements
- [ X ] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [ X ] I linted the code by running `yarn format`
- [ X ] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
Screenshot

### CODE REVIEW

File: websites/0-9/1anime/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.11",
+	"apiVersion": 1,
+	"author": {
+		"id": "331606403057713154",
+		"name": "faetalize"
+	},
+	"service": "1Anime",
+	"description": {
+		"en": "A presence integrating the famous 1anime streaming website!"
```
Comment (theusaf) [inline websites/0-9/1anime/metadata.json:10]: For description, use the official description from the website

File: websites/0-9/1anime/presence.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+const presence = new Presence({
+		clientId: "1314062632419852309",
+	}),
+	strings = presence.getStrings({
+		play: "presence.playback.playing",
+		pause: "presence.playback.paused",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
```
Comment (theusaf) [inline websites/0-9/1anime/presence.ts:10]:
```suggestion
const enum Assets {
```

File: websites/0-9/1anime/presence.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+				const title = document.querySelector(
+						"div#details h3.font-Archivo"
+					).textContent,
+					splitIndex = title.lastIndexOf(" - ");
+
+				//not needed for now
+				// const episodeName : string = document
+				// 	.querySelector("div#details div.flex.font-Archivo.font-bold")
+				// 	.textContent;
+
```
Comment (theusaf) [inline websites/0-9/1anime/presence.ts:42]:
```suggestion
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/0-9/1anime/metadata.json (hunk 1 above)
Comment (faetalize) [reply websites/0-9/1anime/metadata.json]: Done.

**Review by theusaf — APPROVED** [review]

File: websites/0-9/1anime/presence.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+			if (document.location.pathname === "/") {
+				presenceData.details = "Checking out 1anime's homepage!";
+				presenceData.startTimestamp = browsingTimestamp;
+			} else if (document.location.pathname.includes("/anime/watch/")) {
+				//player state ig
+				const player = document.querySelector("video"),
+					timestamps = presence.getTimestamps(
+						player.currentTime,
+						player.duration
+					);
```
Comment (Bas950) [inline websites/0-9/1anime/presence.ts:34]:
```suggestion
					timestamps = presence.getTimestampsFromMedia(player);
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload proof of it working

File: websites/0-9/1anime/presence.ts (hunk 4 above)
Comment (Dev1Anim) [reply websites/0-9/1anime/presence.ts:34]: It works

**Comment by faetalize** [comment]
> Upload proof of it working

Check the original post, I uploaded a screenshot.

**Review by Bas950 — APPROVED** [review]

**Pull request merged**